File: app/src/main/java/expressive/fan/core/MainActivityController.java
package expressive.fan.core;

import expressive.fan.states.State;

public interface MainActivityController {
    void playClicked();
    void pauseClicked();
    void stopClicked();

    void createPlayer();
    void play();
    void pause();
    void stop();

    void setPlayButtonEnabled(boolean b);
    void setPauseButtonEnabled(boolean b);
    void setStopButtonEnabled(boolean b);

    void setNextState(State nextState);

    void progressChanged(int progress, boolean fromUser);

    void startScheduler();

    void updateProgress();

    void resetProgress();

    void seekToNewPos(int progress);

    void completed();
}

File: app/src/main/java/expressive/fan/core/MainActivityController.kt
package expressive.fan.core

import expressive.fan.states.State

interface MainActivityController {
    fun playClicked()
    fun pauseClicked()
    fun stopClicked()
    fun createPlayer()
    fun play()
    fun pause()
    fun stop()
    fun setPlayButtonEnabled(b: Boolean)
    fun setPauseButtonEnabled(b: Boolean)
    fun setStopButtonEnabled(b: Boolean)
    fun setNextState(nextState: State)
    fun progressChanged(progress: Int, fromUser: Boolean)
    fun startScheduler()
    fun updateProgress()
    fun resetProgress()
    fun seekToNewPos(progress: Int)
    fun completed()
}